servicetalk-concurrent-api-internal/src/main/java/io/servicetalk/concurrent/api/internal/SpliceFlatStreamToSingleResult.java

```diff
@@ -259,47 +257,58 @@ protected void handleSubscribe(PublisherSource.Subscriber<? super Payload> newSu
                     } else {
-                        ((PublisherSource.Subscriber<Payload>) maybeSubscriber).onError(new IllegalStateException(
+                        ((PublisherSource.Subscriber<Following>) maybeSubscriber).onError(new IllegalStateException(
                                 "Duplicate Subscribers are not allowed. Existing: " + maybeSubscriber +
                                         ", failed the race with a duplicate, but neither has seen onNext()"));
                     }
                 } else {
-                    LOGGER.debug("Terminal error queued for delayed delivery to the payload publisher. " +
-                            "If the payload is not subscribed, this event will not be delivered.", t);
+                    LOGGER.debug("Terminal error queued for delayed delivery to the Publisher<Following>. " +
+                            "If it is not subscribed, this event will not be delivered.", t);
                 }
             }
         }
 
         @SuppressWarnings("unchecked")
         @Override
         public void onComplete() {
-            if (payloadSubscriber != null) { // We have a subscriber that has seen onNext()
-                payloadSubscriber.onComplete();
+            if (followingSubscriber != null) { // We have a subscriber that has seen onNext()
+                followingSubscriber.onComplete();
             } else {
-                final Object maybeSubscriber = maybePayloadSubUpdater.getAndSet(this, EMPTY_COMPLETED);
+                final Object maybeSubscriber = maybeFollowingSubUpdater.getAndSet(this, EMPTY_COMPLETED);
                 if (maybeSubscriber instanceof PublisherSource.Subscriber) {
-                    if (maybePayloadSubUpdater.compareAndSet(SplicingSubscriber.this, EMPTY_COMPLETED,
-                            EMPTY_COMPLETED_DELIVERED)) {
-                        ((PublisherSource.Subscriber<Payload>) maybeSubscriber).onComplete();
+                    if (maybeFollowingSubUpdater.compareAndSet(this, EMPTY_COMPLETED, EMPTY_COMPLETED_DELIVERED)) {
+                        ((PublisherSource.Subscriber<Following>) maybeSubscriber).onComplete();
                     } else {
-                        ((PublisherSource.Subscriber<Payload>) maybeSubscriber).onError(new IllegalStateException(
+                        ((PublisherSource.Subscriber<Following>) maybeSubscriber).onError(new IllegalStateException(
                                 "Duplicate Subscribers are not allowed. Existing: " + maybeSubscriber +
                                         ", failed the race with a duplicate, but neither has seen onNext()"));
                     }
-                } else if (!metaSeenInOnNext) {
-                    dataSubscriber.onError(new IllegalStateException("Empty stream"));
+                } else if (!firstSeenInOnNext) {
+                    ensureResultSubscriberOnSubscribe();
+                    resultSubscriber.onSuccess(null);
```

Comment:
`Single` completes with `null` instead of `IllegalStateException("Empty stream")` when the original publisher is empty.